spec/features/update_username_spec.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'Update username controller', type: :feature, js: true do
+  let!(:user) { create(:user, id: 500, username: 'Temp') }
+
+  before do
+    login_as user
+  end
+
+  context 'Update Username page' do
+    it 'visits and enter updated username' do
+      VCR.use_cassette('update_username') do
+        create(:user, id: 1, username: 'Old Username', global_id: '14093230')
```

Comment:
It's better to avoid setting arbitrary values that don't matter to the test, in this case `id`. You can find the user in the `expect` statement below via global_id instead.